apps/api/src/provider/services/provider/provider.service.ts

```
@@ -140,13 +146,82 @@ export class ProviderService {
       ]
     });
 
-    const distinctProviders = providersWithAttributesAndAuditors.filter((value, index, self) => self.map(x => x.hostUri).lastIndexOf(value.hostUri) === index);
+    const distinctProviders = Object.values(
+      providersWithAttributesAndAuditors.reduce((acc: Record<string, Provider>, provider: Provider) => {
+        return {
+          ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)